src/sync/test.rs

```diff
@@ -205,12 +203,10 @@ fn transactions() {
         .expect("session creation should succeed");
     let coll = init_db_and_typed_coll(&client, function_name!(), function_name!());
 
-    if should_create_collection {
-        client
-            .database(function_name!())
-            .create_collection(function_name!(), None)
-            .expect("create collection should succeed");
-    }
+    client
```

Comment:
A few errors were showing up on 4.4 replica sets for this test with the following message: "Unable to read from a snapshot due to pending collection catalog changes; please retry the operation." It seems like this has caused some [drivers issues](https://jira.mongodb.org/browse/SERVER-41532) in the past related to inserts in transactions. Creating the collection before starting the transaction fixes the issue.